Kerberos.NET/Dns/IKerberosDnsQuery.cs

```diff
@@ -0,0 +1,24 @@
+﻿// -----------------------------------------------------------------------
+// Licensed to The .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// -----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Kerberos.NET.Dns
+{
+    /// <summary>
+    /// Provides a mechanism to query for DNS records as used by Kerberos.
+    /// </summary>
+    public interface IKerberosDnsQuery
+    {
+        /// <summary>
+        /// Make a DNS lookup for the provided query and record type.
+        /// </summary>
+        /// <param name="query">The query to send to the DNS server.</param>
+        /// <param name="type">The requested record type of the query.</param>
+        /// <returns>Returns zero or more results from the query.</returns>
+        Task<IEnumerable<DnsRecord>> Query(string query, DnsRecordType type);
```

Comment:
```suggestion
        Task<IReadOnlyCollection<DnsRecord>> Query(string query, DnsRecordType type);
```
? 

In general an IEnumerable can be iterated multiple times. Returning it by a task, could result in unexpected behavior (e.g. if the source of the enumerable got disposed, etc.). 
Returning a collection (already materialized) is safe, as the content are "finished".

I see how it's implemented -- for win it's actually a `List<DnsRecord>` -- but as this is a public API, hence the surface should be in proper state.